### ISSUE TITLE
Type error when using middleware together with .doc method with @hono/zod-openapi

### ISSUE THREAD

**Issue by kobeindemans** [honojs/middleware#246]
I would like to combine middleware functions with OpenAPI documentation but get the following type error:

`Property 'doc' does not exist on type 'Hono<Env, {}, "/">'.ts(2339)
`

My code is as follows:
```typescript
import { OpenAPIHono } from "@hono/zod-openapi";
import { cors } from "hono/cors";

const app = new OpenAPIHono().use("/*", cors());

// api documentation
app.doc("/doc", {
  openapi: "3.0.0",
  info: {
    version: "1.0.0",
    title: "My API",
  },
});

export default app;
```

**Comment by yusukebe** [comment]
Hi @KobeIndemans,

It's possible that the versions of Hono used in `@hono/zod-openapi` and `hono` are different. Could you upgrade them to the latest versions?

**Comment by yudai-nkt** [comment]
I suppose this is the same as honojs/hono#1529.

**Comment by yusukebe** [comment]
I don't think this is related to https://github.com/honojs/hono/issues/1529. With this simple approach, we can use Hono's middleware in `OpenAPIHono`.

**Comment by dinfer** [comment]
Same problem here.
The `use` method is not overrided by `OpenAPIHono` class, so, as Hono declares, it returns an instance of `BaseHono`, rather than `OpenAPIHono`. The type is lost, any mehod of `OpenAPIHono` cannot be used since the `use` call.
It's a bug of `@hono/zod-openapi`, not the `hono` itself.

**Comment by dinfer** [comment]
The variable `app` is not an instance of `OpenAPIHono`, So I got an error: `Property 'doc' does not exist on type 'Hono<{}, {}, "/", "*">'`

<img width="906" height="438" alt="Image" src="https://github.com/user-attachments/assets/1403c1f2-9712-4511-86cd-7bd47a75e771" />

I'm adding a type override like this:

<img width="1760" height="900" alt="Image" src="https://github.com/user-attachments/assets/53736ba2-9076-463c-a01f-ca7e7d7b9e87" />

Everything is ok

<img width="846" height="470" alt="Image" src="https://github.com/user-attachments/assets/c727a523-a2a1-4d6e-ae0b-23bb08330f0d" />

Package information:
```
├── @hono/zod-openapi@1.2.0
├── @types/bun@1.3.5
├── hono@4.11.2
└── typescript@5.9.3
```

**Comment by yusukebe** [comment]
Hi @dinfer 

We created the `$` function to solve the problem. Please try it.

https://github.com/honojs/middleware/tree/main/packages/zod-openapi#type-utilities

**Comment by dinfer** [comment]
@yusukebe Thanks! I’m curious why we need to add a wrapper here—it feels a bit more tedious. The reason I love Hono is its simplicity, after all.

**Comment by yusukebe** [comment]
Hi @dinfer

Yeah. I really want to make it simpler because Hono should be simple, as you said. But if I define the type you suggest, type errors are thrown in the source file.

<img width="1484" height="528" alt="Image" src="https://github.com/user-attachments/assets/3080ca82-e583-4e1f-8a49-a0750788034a" />

I think the type definition of `app.use` is too complex. So using the wrapper makes sense.
